osu.Game/Online/Notifications/NotificationsClientConnector.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Game.Online.API;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Online.Notifications
+{
+    /// <summary>
+    /// An abstract connector or <see cref="NotificationsClient"/>s.
+    /// </summary>
+    public abstract class NotificationsClientConnector : PersistentEndpointClientConnector
+    {
+        public event Action<Channel>? ChannelJoined;
+        public event Action<List<Message>>? NewMessages;
+        public event Action? PresenceReceived;
+
+        private bool chatStarted;
+
+        protected NotificationsClientConnector(IAPIProvider api)
+            : base(api)
+        {
+        }
+
+        public void StartChat()
```

Comment:
I don't understand why this is a thing. Is there a case we *wouldn't* enable chat?